### ISSUE TITLE
A parent modelAction that calls a modelAction inside a child with undoMiddleware throws: patchRecorderData is undefined

### ISSUE THREAD

**Issue by Phault** [xaviergonz/mobx-keystone#168]
Sorry for the cryptic title. I have a RootStore which contains another model SubStore. The SubStore has some prop and a modelAction that changes that prop. The RootStore also has a modelAction that just calls the SubStore's modelAction. This all works fine and dandy but an issue arises when the SubStore is wrapped in undoMiddleware. If I call the SubStore's modelAction then everything is fine but if I instead call the RootStores modelAction then it explodes complaining that patchRecorderData is undefined.
If the RootStore modelAction directly changes the SubStore's prop instead of using its modelAction, then everything works fine as well.

[I've made a codesandbox that showcases the issue here](https://codesandbox.io/s/mobx-keystone-undomiddleware-nested-modelactions-diq2i?file=/src/index.ts).

**Comment by xaviergonz** [comment]
Thanks, will be fixed in 0.45.1
Note however that if you don't call an action in the "undoable" sub store (e.g. 
```
@modelAction
  addSubStoreValueIndirect(value: number) {
    this.substore.addValue(value);
  }
```
that won't be added to the undo list.

**Comment by xaviergonz** [comment]
Released
